### DISCUSSION TITLE
Filament sensor not working correctly

### DISCUSSION THREAD

**Discussion by jgabb005** [mriscoc/Ender3V2S1#1200] [Q&A]
Just installed this firmware last night. Everything seems to work except for my filament runout sensor. It was working on the previous firmware so I know its wired correctly. 

So here we go

Bug Description
Runout sensor does not pause print.

EXPECTED BEHAVIOR
When sensor has filament running through End Stop Diag displays TRIIGGERED, and when the filament has runout and sensor no longer has filament running through it is OPEN. 

When filament sensor switches from TRIGGERED to OPEN(aka filament has ran out) in the middle of a print, the print will pause immediately for runout distance is set to 0.0

ACTUAL BEHAVIOR
Printer continues to print regardless of the status of the sensor. 

STEPS TO REPRODUCE/DIAGNOSE
Take a 3 inch piece of filament and enter it into the sensor. Check End Stop Diag page and sensor displays TRIGGERED. 

Pull the filament out of the sensor, and sensor displays OPEN

Place 3 inch filament back in sensor, and make sure the extruder has no filament in it. 

Start sample print. Printer begins to start printing as normal. 

Pull up End Stop Diag page. Filament sensor reads TRIGGERED, as it should. Pull filament out of sensor, Filament sensor changes to OPEN, as it should. Printer continues to print.
 
FILAMENT SETTINGS 
Enable Runout- ACTIVE  
Runout Active- LOW 
Runout Distance- 0.0
Invert Extruder- INACTIVE
Extruder min temp- 180
Load mm- 0.0
Unload mm- 100.0 

PRINTER 
Ender 3 V2 w/Sprite extruder

FILAMENT
Nylon X and PL

RUNOUT SENSOR
Creality Ender 3 V2 Filament Runout Sensor
https://www.amazon.com/dp/B099ZT1KNY?ref=ppx_yo2ov_dt_b_product_details&th=1

FIRMWARE  
Ender3V2-422-BLTUBL-MPC-20231217.bin  

Add-ons
Sprite Extruder, dual z, Ender Extender 600mm

Bed Leveling
BL Touch

Slicer
Creality Slicer

**Comment by kjeld96** [comment]
i have the same issue

**Comment by borgesp09** [comment]
same issue. Filament runout enable, end stop diagnostic says it's triggered but print continues right along.

**Reply by mriscoc** [reply]
https://github.com/mriscoc/Ender3V2S1/discussions/1367#discussioncomment-11087968
